### ISSUE TITLE
WebClient PrematureCloseException: Connection prematurely closed BEFORE response

### ISSUE THREAD

**Issue by sylvain-rouquette** [reactor/reactor-netty#756]
Throught it was related to this issue:
https://github.com/spring-projects/spring-framework/issues/22464#issuecomment-499474686

### Expected behavior
When using the WebClient to get binary content in parallel, I should be getting the byte[] without error.

### Actual behavior
Some WebClient fail with an exception, the connection has been closed prematurely.

### Steps to reproduce

I created a project here to reproduce the problem:
https://github.com/InceptoMedical/orthanc-netty

The problem may be related to a bug in Orthanc, I increased the number of threads to 500, but the problem still occurs.
I couldn't reproduce the problem on nginx for example.

### workaround:
Instead of doing WebClient.create(url), use the builder and create a new connection:

```
WebClient.builder()
                .clientConnector(new ReactorClientHttpConnector(HttpClient.newConnection().compress(true)))
                .baseUrl(url).build()
```

### Reactor Netty version
0.8.8.RELEASE

### JVM version (e.g. `java -version`)
11.0.2

### OS version (e.g. `uname -a`)
Windows 10 Pro

**Comment by violetagg** [comment]
@sylvain-rouquette 
Can you please run wireshark in order to check the communication between the server and the client.

Before that please do the following change:

```
    private Flux<String> getInstanceList() {
        return WebClient.builder()
                .clientConnector(new ReactorClientHttpConnector(HttpClient.create().compress(true).wiretap(true)))
                .baseUrl(INSTANCE_URL).build()
                .get()
....

    private Mono<byte[]> getContent(String instance) {
        String url = String.format("%s/%s/file", INSTANCE_URL, instance);
        return WebClient.builder()
                .clientConnector(new ReactorClientHttpConnector(HttpClient.create().compress(true).wiretap(true)))
                .baseUrl(url).build()
                .get()
```

Using `wiretap` we will see the communication between the server and the client also in the logs.
(you will need `application.properties` with `logging.level.reactor.netty=debug`)


Then from the logs search for `Connection prematurely closed BEFORE response` in order to obtain the client port. Then check the communication in the wireshark for this port. It is interesting how server responds. It may appear that there is an issue with the server.

**Comment by violetagg** [comment]
@sylvain-rouquette Closing this as there is not enough information.

**Comment by sylvain-rouquette** [comment]
sure, I'll update it when I have more time. But 2 weeks and closing the bug is a bit short.

**Comment by davesargrad** [comment]
> sure, I'll update it when I have more time. But 2 weeks and closing the bug is a bit short.

I agree with @sylvain-rouquette . I am also seeing this, problem, and I see other GIT reports that seem to pertain to the use of the netty http library that suggest this is a real bug. 

Strikes me as odd that this would be closed and marked as not a bug.

![image](https://user-images.githubusercontent.com/48793689/86584680-4770ff80-bf53-11ea-96e0-9b7f15da20e1.png)

**Comment by violetagg** [comment]
@davesargrad we need at least a tcp dump in order to proceed with the investigation. When such is provided we can try to identify the issue. For example this issue #1170 or this issue #1092
Closing a bug does not mean that we cannot reopen it when more information is available.
Also we ask people to open a new issue when they face this exception because even that the exception seems to be the same, sometime the root cause is totally different. Examples - the issues above + the following 
https://github.com/reactor/reactor-netty/issues/1008
https://github.com/reactor/reactor-netty/issues/1038
https://github.com/reactor/reactor-netty/issues/564#issuecomment-576244256
